game_tic_tac_toe.py

```diff
@@ -159,11 +184,130 @@ def player_won(self, check_disable):
+        '''In this function the computer decides the next move
+        Parameters:
+        disable_numbers_o_copy(set): copy of the moves up to this moment of the computer
+        disable_numbers_x_copy(set): copy of the moves up to this moment of the player
+        Return
+        move(int): position on the board'''
+        disable_numbers_o_copy = self.disable_numbers_o.copy()
+        disable_numbers_x_copy = self.disable_numbers_x.copy()
+
+        available_moves = []
+        for i in range(9):
+            if self.game_btns[i]['state'] == 'normal':
+                available_moves.append(i)
+        print(f'available moves for the computer {available_moves}')
+        
+        # identify computer player and move done by him 
+        # and by its opponent
+        if comp_symbol == "x":
+            computer_set = disable_numbers_x_copy
+            oponent_set = disable_numbers_o_copy
+        else:
+            computer_set = disable_numbers_o_copy
+            oponent_set = disable_numbers_x_copy
+        print(f'computer already played - disable numbers {computer_set}')
+
+        # Check for every available move if computer can win with the next move
+        # if it can win self.move has a value
+        for i in available_moves:
+            computer_set.add(i)
+            if self.player_won(computer_set):
+                move = i
+                print(f'computer can win!! {move}')
+                return move
+            computer_set.remove(i)
+        print('computer does not have a chance to win')
+
+        # Check for every available move if the human can win with the next move
+        # if the human could win next round - the computer will block him/her
+        for i in available_moves:
+            oponent_set.add(i)
+            if self.player_won(oponent_set):
+                move = i
+                print(f'opponent can win!! block: {move}')
+                return move
+            oponent_set.remove(i)
+        print('oponent has no chance to win')
+
+        # Check if any of the corners and center are available to make a move
+        # Center and corners are python-positions [0,2,6,8,4] in the board
+        corners_center_available = []
+        for i in available_moves:
+            if i in [0, 2, 4, 6, 8]:
+                corners_center_available.append(i)
+        if corners_center_available:
+            move = random.choice(corners_center_available)
+            print(f'center or corner has been choosen: {move}')
+            return move
+
+        print('no corners available')
+
+        # Check if any of the sides are available to make a move
+        # Center and corners are python positions [1,3,5,7] in the board
+        sides_available = []        
+        for i in available_moves:
+            if i in [1, 3, 5, 7]:
+                sides_available.append(i)
+        if sides_available:
+            move = random.choice(sides_available)
+            return move
```

Comment:
At this point you can just check for available moves because you already know that all the good moves are not possible: `move = random.choice(available_moves)`